tests/Unit/Http/RuleMatcherTest.php

```diff
@@ -15,14 +15,15 @@
 use FOS\HttpCacheBundle\Http\RuleMatcher;
 use PHPUnit\Framework\TestCase;
 use Symfony\Component\HttpFoundation\Request;
-use Symfony\Component\HttpFoundation\RequestMatcher;
+use Symfony\Component\HttpFoundation\RequestMatcher\AttributesRequestMatcher;
+use Symfony\Component\HttpFoundation\RequestMatcherInterface;
 use Symfony\Component\HttpFoundation\Response;
 
 class RuleMatcherTest extends TestCase
 {
     public function testRequestMatcherCalled()
     {
-        $requestMatcher = new RequestMatcher(null, null, null, null, ['_controller' => '^AcmeBundle:Default:index$']);
+        $requestMatcher = new AttributesRequestMatcher(['_controller' => '^AcmeBundle:Default:index$']);
```

Comment:
ah right, we need to fix those for symfony 7.

the new class is only available since Symfony 6.2 can you please do `class_exists(AttributesRequestMatcher::class) ? new AttributesRequestMatcher(...) : new RequestMatcher(...);` to keep legacy tests working?

and i would prefer this in a separate pull request as it is not related to extra framework bundle. it is fixing a deprecation in symfony 6 and avoid the error in symfony 7.